app/helpers/discovered_hosts_helper.rb

```diff
@@ -94,8 +94,15 @@ def provision_button(host, authorization_options)
   end
 
   def host_path(host)
-    return super unless controller_name == 'discovered_hosts'
-    discovered_host_path(host)
+    return discovered_host_path(host) if controller_name == 'discovered_hosts'
+    # This is a hack to fix the hostgroup selection on the discovered host edit page
+    # The hostgroup onChange action is replacing the form url from `/discovered_hosts/:id`
+    # to `/hosts/:id``, which is not correct (in this case) and it breaks the form submission
+    return discovered_host_path(host) if controller_name == 'hosts' &&
+                                         action_name == 'process_hostgroup' &&
```

Comment:
While I'm OK with specifying this particular method as it's the one that causes the bug, I wonder if we shouldn't include all the AJAX methods from the [controller](https://github.com/theforeman/foreman/blob/2774da229db941d1f0b55c92ccab8935951397a4/app/controllers/hosts_controller.rb#L15)

Reply:
Maybe, but I'm worried about the impact it can have across the plugins. This fix is right now important for downstream release, can we go with it as it is and later on think about the refactoring? Dont want to block QAs who are waiting for this fix.